modules/tinymce/src/themes/silver/main/ts/ui/core/color/Options.ts

```diff
@@ -84,18 +117,35 @@ const register = (editor: Editor): void => {
   });
 };
 
-const getColorCols = option('color_cols');
+const getColorCols = (editor: Editor, id: string): number => {
+  if (id === foregroundId) {
+    return option('color_cols_foreground')(editor);
+  } else if (id === backgroundId) {
+    return option('color_cols_background')(editor);
+  } else {
+    return option('color_cols')(editor);
+  }
+};
 const hasCustomColors = option('custom_colors');
-const getColors = option<Menu.ChoiceMenuItemSpec[]>('color_map');
+const getColors = (editor: Editor, id: string): Menu.ChoiceMenuItemSpec[] => {
+  if (id === foregroundId && editor.options.isSet('color_map_foreground')) {
+    return option<Menu.ChoiceMenuItemSpec[]>('color_map_foreground')(editor);
+  } else if (id === backgroundId && editor.options.isSet('color_map_background')) {
+    return option<Menu.ChoiceMenuItemSpec[]>('color_map_background')(editor);
+  } else {
+    return option<Menu.ChoiceMenuItemSpec[]>('color_map')(editor);
+  }
+};
 
-const getCurrentColors = (): Menu.ChoiceMenuItemSpec[] => Arr.map(colorCache.state(), (color) => ({
-  type: 'choiceitem',
-  text: color,
-  value: color
-}));
+const getCurrentColors = (id: string): Menu.ChoiceMenuItemSpec[] =>
+  Arr.map(getCatcheForId(id).state(), (color) => ({
+    type: 'choiceitem',
+    text: color,
+    value: color
+  }));
 
-const addColor = (color: string): void => {
-  colorCache.add(color);
+const addColor = (id: string, color: string): void => {
```

Comment:
I don't think this should live in the options module. Since all it does is to add a color to the cache.